tests/integration/v1/test_integration.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import shutil
+
+logger = logging.getLogger(__name__)
+
+LIB_DIR = "lib/charms/tls_certificates_interface/v1/tls_certificates.py"
+REQUIRER_CHARM_DIR = "tests/integration/v1/requirer_charm"
+PROVIDER_CHARM_DIR = "tests/integration/v1/provider_charm"
+TLS_CERTIFICATES_PROVIDER_APP_NAME = "tls-certificates-provider"
+TLS_CERTIFICATES_REQUIRER_APP_NAME = "tls-certificates-requirer"
+
+
+def copy_lib_content() -> None:
+    shutil.copyfile(src=LIB_DIR, dst=f"{REQUIRER_CHARM_DIR}/{LIB_DIR}")
+    shutil.copyfile(src=LIB_DIR, dst=f"{PROVIDER_CHARM_DIR}/{LIB_DIR}")
+
+
+class TestIntegration:
+    requirer_charm = None
+    provider_charm = None
+
+    async def test_given_charms_packed_when_deploy_charm_then_status_is_blocked(self, ops_test):
+        copy_lib_content()
+        TestIntegration.requirer_charm = await ops_test.build_charm(f"{REQUIRER_CHARM_DIR}/")
+        TestIntegration.provider_charm = await ops_test.build_charm(f"{PROVIDER_CHARM_DIR}/")
```

Comment:
The charms are being built in the test itself, why do we say `given_charms_packed`, the test can actually fail because of this reason, doesn't it?

Reply:
Yes the "given" is often in the test itself but that's not the object of the test. Here we're not testing that the charms get packed, it's just a pre-requisite for testing something else.